game/scripts/vscripts/abilities/stopfightingyourself/dupe_heroes.lua

```diff
@@ -0,0 +1,145 @@
+
+require('libraries/timers')
+
+LinkLuaModifier("modifier_boss_stopfightingyourself_illusion", "abilities/stopfightingyourself/dupe_heroes.lua", LUA_MODIFIER_MOTION_NONE)
+
+
+boss_stopfightingyourself_dupe_heroes = class({})
+
+function boss_stopfightingyourself_dupe_heroes:GetAOERadius()
+  	return self:GetSpecialValueFor('radius')
+end
+
+--[[
+Credits: Pizzalol
+]]
+function boss_stopfightingyourself_dupe_heroes:OnSpellStart()
+  local caster = self:GetCaster()
+  local target = caster:GetAbsOrigin()
+
+  if caster.illusions and #caster.illusions >= self:GetSpecialValueFor('max_illusions') then
+    return
+  end
+
+  local units = FindUnitsInRadius(
+    caster:GetTeamNumber(),
+    target,
+    nil,
+    self:GetSpecialValueFor('radius'),
+    self:GetAbilityTargetTeam(),
+    self:GetAbilityTargetType(),
+    self:GetAbilityTargetFlags(),
+    FIND_ANY_ORDER,
+    false
+  )
+
+
+  for _,unit in ipairs(units) do
+    if unit:IsRealHero() and unit ~= caster then
+      self:StartCooldown(self:GetSpecialValueFor('cooldown'))
+
+      local illusion = CreateUnitByName(
+        unit:GetName(),
+        unit:GetAbsOrigin(),
+        true,
+        caster,
+        caster,
```

Comment:
[This](http://moddota.com/forums/discussion/62/illusion-ability-example) says that the unit owner should be `nil`, but since this doesn't seem to crash the game, it could be outdated information.